src/material/slider/slider.ts

```diff
@@ -636,21 +645,26 @@ export class MatSlider extends _MatSliderMixinBase
     }
 
     this._ngZone.run(() => {
-      const oldValue = this.value;
-      const pointerPosition = getPointerPositionOnPage(event);
-      this._isSliding = true;
-      this._lastPointerEvent = event;
-      event.preventDefault();
-      this._focusHostElement();
-      this._onMouseenter(); // Simulate mouseenter in case this is a mobile device.
-      this._bindGlobalEvents(event);
-      this._focusHostElement();
-      this._updateValueFromPosition(pointerPosition);
-      this._valueOnSlideStart = oldValue;
-
-      // Emit a change and input event if the value changed.
-      if (oldValue != this.value) {
-        this._emitInputEvent();
+      this._touchId = isTouchEvent(event) ?
+        event.touches[event.touches.length - 1]?.identifier : undefined;
```

Comment:
I think that this doesn't work as intended, if the touches are not all released in the opposite order as they are started

Reply:
This is only the `touchstart` though, after this the order of events doesn't matter.

Reply:
yes, but here, we get the touch id. I'm not sure if your assumption, that the last touch in the touches list is "our" touch, is correct.

Reply:
I added a console.log(this._touchId); in line 650 to verify the ID, and when I am repeating my steps from my comment [1], I get the output:
0
1
1
That means, the middle slider listens on the same ID as the right slider

[1] https://github.com/angular/components/pull/22615#issuecomment-830833842

Reply:
When I was working on it, I tried logging out the identifiers and it seemed like they were being added sequentially any time you tap with a different finger.

Reply:
My observation is, that the identifiers are always in sorted order in the array.
If the touch 0 is finished, but touch 1 still goes on, and a new touch starts, it gets the ID 0 and this then on position 0 in the array
- touch start 
- [0]
- touch start
- [0, 1]
- touch end (0)
- [1]
- touch start
- [0, 1]

Reply:
Hmm, that's weird. We might be able to get away with checking the `target` here since the `touchstart` even is bound on the slider itself.

Reply:
yes, good point!

Reply:
I logged the event.target and also the event.touches[].target at line 650 (_pointerDown)
Depending on where I start the touch, I get different Elements as target:
* <div class="mat-slider-ticks">
* <div class="mat-slider-thumb">
* <mat-slider>
Not sure if there are more, but it should always be mat-slider or a child of it.
But, the event.target is always the one of the latest touch down, so we **could** find the right target like that:
` this._touchId = isTouchEvent(event) ? getLatestTouchId(event) : undefined; ` 
with: 
`function getLatestTouchId(event: TouchEvent): number | undefined {
    for (let i = 0; i < event.touches.length; i++) {
        if (event.touches[i].target === event.target) {
            return event.touches[i].identifier;
        }
    }
    return undefined;
}


`

Reply:
Perhaps some advice from someone who has a deep knowledge on multi touch events on different devices would be helpful.
Is there someone who could be tagged?

Reply:
I just looked at some code which I implemented for my job. It's simulating a touch device (uinput) on linux.
There, I also have to define the identifier that is used for the touch.

My assumption (which I verify with my code from work) is, that the identifier in the touch event (js in the browser) is given from the touch device (driver) and not from the browser.

That would then explain why it works differently on android and the iPhone

Reply:
Thank you for looking into it, I'll go with a modified version of your proposal for now and we can iterate on it if it ends up breaking something. 

Also yeah, it's weird that there isn't a standard way for the identifiers to be generated. I noticed that the Chrome dev tools has a different way of doing it too.

Reply:
I'll test your implementation this evening.

Reply:
Okay, I've tested on android with firefox and chrome.
On both devices, multitouch works now as expected, including the scenario that was not working before.

Great!
Thanks a lot!

Reply:
yay, thank you for verifying 🎉